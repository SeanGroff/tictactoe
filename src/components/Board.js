import React from 'react';
import styled from 'styled-components';

// Placeholder board for initial state
const board = ['', '', '', '', '', '', '', '', ''];

const GameBoard = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  max-width: 150px;
`;

const Cell = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50px;
  height: 50px;
  border-color: black;
  border-style: solid;
`;

const Board = props => {
  return (
    <GameBoard>
      {board.map((cell, index) => (
        <Cell id={`cell-${index}`} position={index} key={index}>{cell}</Cell>
      ))}
    </GameBoard>
  );
};

export default Board;
