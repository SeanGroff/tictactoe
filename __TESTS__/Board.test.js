import React from 'react';
import ReactDOM from 'react-dom';
import Board from '../src/components/Board';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Board />, div);

  const component = renderer.create(<Board />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
