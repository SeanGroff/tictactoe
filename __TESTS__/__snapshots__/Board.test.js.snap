exports[`test renders without crashing 1`] = `
<div
  className="bcTBmH">
  <div
    className="igXHqs"
    id="row-0">
    <span
      className="kNuHNb"
      id="col-0-0">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
    <span
      className="kNuHNb"
      id="col-0-1">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
    <span
      className="kNuHNb"
      id="col-0-2">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
  </div>
  <div
    className="igXHqs"
    id="row-1">
    <span
      className="kNuHNb"
      id="col-1-0">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
    <span
      className="kNuHNb"
      id="col-1-1">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
    <span
      className="kNuHNb"
      id="col-1-2">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
  </div>
  <div
    className="igXHqs"
    id="row-2">
    <span
      className="kNuHNb"
      id="col-2-0">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
    <span
      className="kNuHNb"
      id="col-2-1">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
    <span
      className="kNuHNb"
      id="col-2-2">
      <div
        className="hdKDHS"
        onClick={[Function]} />
    </span>
  </div>
</div>
`;
